src/iterated_integrals.jl

```diff
@@ -0,0 +1,345 @@
+    @unpack m_seq = Wik
+    m      = length(dW)
+    M      = Int(m*(m-1)/2)
+    sum_dW² = dW'*dW
+
+    WikJ = dW*dW'
+    Gp₁ = randn(M)
+    α = sqrt(1 + sum_dW²/dt)
+    Gp₂ = Gp₁/(sqrt(2)*(1+α)*dt)
+
+    #operator (Iₘ² - Pₘ)Kₘᵀ
+    for i in 1:M
+        WikJ2[m_seq[i,1], m_seq[i,2]] = Gp₂[i]
+        WikJ2[m_seq[i,2], m_seq[i,1]] = -Gp₂[i]
+    end
+
+    #operator (Iₘ X W*Wᵀ)
+    WikJ2 = WikJ*WikJ2
+
+    #operator Kₘ(Iₘ² - Pₘ)
+    WikJ2 = WikJ2 - WikJ2'
+    for i in 1:M
+        Gp₂[i] = WikJ2[m_seq[i,1], m_seq[i,2]]
+    end
+    Gp₁ = Gp₁/sqrt(2) + Gp₂
+
+    #operator (Iₘ² - Pₘ)Kₘᵀ
+    for i in 1:M
+        WikJ2[m_seq[i,1], m_seq[i,2]] = Gp₁[i]
+        WikJ2[m_seq[i,2], m_seq[i,1]] = -Gp₁[i]
+    end
+
+    WikJ *= 1//2
+    𝑎ₚ = (π^2)/6
+    p = Int(floor((1/(C*π))*sqrt(M/(24*dt))*sqrt(m + 4*sum_dW²/dt) + 1))
+    Aᵢ = false .* vec(dW)   # Aᵢ is vector of aᵢ₀
+    for r in 1:p
+        𝑎ₚ -= (1/r^2)
+        var = sqrt(dt/(2*π*r))
+        vec_ζ = randn(m)*var
+        vec_η = randn(m)*var
+        WikJ += (vec_ζ*vec_η' - vec_η*vec_ζ')
+        Aᵢ -= (2/sqrt(π*r))*vec_ζ
+    end
+
+    WikJ -= 1//2*(dW*Aᵢ' - Aᵢ*dW')
+    WikJ += (sqrt(𝑎ₚ)*dt/π)*WikJ2
+    WikJ
+end
+
+"""
+
+    get_iterated_I!(dt, dW, Wik::WikJGeneral_iip, C=1)
+
+This function calculates WikJ, a mxm Array for a m dimensional general noise problem, which is a approximation
+to the second order iterated integrals.
+
+WikJDiagonal and WikJCommute use the properties of respective noises to simplify the calculations.
+While the calculation for General Noise case is taken from section 4 of [SDELab: A Package for solving stochastic differential
+equations in MATLAB](https://doi.org/10.1016/j.cam.2006.05.037) and SDELAB2(https://github.com/tonyshardlow/SDELAB2)
+which is the Implementation of SDELab in Julia.
+```math
+    𝒜ᵖ = (Iₘ² - Pₘ)Kₘᵀ Δt/(2π) √(𝑎ₚ) √(Σ∞) Gp₁
+```
+
+```math
+    √(Σ∞) = (Σ∞ + 2αIₘ)/(√2 * (1 + α))
+```
+
+let the combined operators be,
+```math
+    F = Kₘ(Iₘ² - Pₘ)(Iₘ ⨂ W(Δt)W(Δt)ᵀ)(Iₘ² - Pₘ)Kₘᵀ
+```
+
+```math
+    Σ∞ = 2Iₘ + (2/Δt)F
+```
+
+See the paper for further details of specific operators.
+Here we've only shown in which order these are implemented in this code.
+
+From above we can see:
+
+```math
+    Δt/(2π) √(𝑎ₚ) √(Σ∞) Gp₁ = Δt/π √(𝑎ₚ) (√(Σ∞)/2 Gp₁)
+```
+
+let,
+```math
+    Oper2(Gp₁) = (√(Σ∞)/2 Gp₁) = (Iₘ/√2 + F/(√2 * (1 + α) * Δt))(Gp₁)
+    Oper2(Gp₁) = Iₘ*Gp₁/√2 + F(Gp₁/(√2*(1+α)*Δt))
+    Oper2(Gp₁) = Gp₁/√2 + F(Gp₁/(√2*(1+α)*Δt))
+```
+
+we have,
+
+    Gp₂ = Gp₁/(sqrt(2)*(1+α)*dt)
+
+```math
+    𝒜ᵖ = (Iₘ² - Pₘ)Kₘᵀ Δt/π √(𝑎ₚ) Oper2(Gp₁/√2 + F(Gp₂))
+    𝒜ᵖ = √(𝑎ₚ)*Δt/π * (Iₘ² - Pₘ)Kₘᵀ Oper2(Gp₁/√2 + F(Gp₂))
+```
+In the code we have
+
+```math
+    WikJ2 = (Iₘ² - Pₘ)Kₘᵀ(Oper2(Gp₁))
+```
+
+"""
+function get_iterated_I!(dt, dW, Wik::WikJGeneral_iip, C=1)
+    @unpack WikJ, WikJ2, WikJ3, m_seq, vec_ζ, vec_η, Gp₁, Gp₂, Aᵢ = Wik
+
+    m      = length(dW)
+    M      = Int(m*(m-1)/2)
```

Comment:
`div(m * (m - 1), 2)` or `m * (m - 1) ÷ 2` would be better, as @devmotion suggested.